editor/components/page-attributes/parent.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import { stringify } from 'querystringify';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withInstanceId, withAPIData } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+import { buildTermsTree } from '@wordpress/utils';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPostId, getEditedPostAttribute } from '../../store/selectors';
+import { editPost } from '../../store/actions';
+import TermTreeSelect from '../../../blocks/term-tree-select/index.js';
+
+export function PageAttributesParent( { parent, postId, postType, items, onUpdateParent } ) {
+	const parentPageLabel = get( postType, 'labels.parent_item_colon' );
+	const pagesTree = buildTermsTree( get( items, 'data', [] ).filter( item => item.id !== postId ).map( ( item ) => ( {
+		id: item.id,
+		parent: item.parent,
+		name: item.title.raw ? item.title.raw : `#${ item.id } (${ __( 'no title' ) })`,
+	} ) ) );
+	return (
+		<TermTreeSelect
+			label={ parentPageLabel }
+			noOptionLabel={ `(${ __( 'no parent' ) })` }
+			termsTree={ pagesTree }
+			selectedTerm={ parent }
+			onChange={ onUpdateParent }
+		/>
+	);
+}
+
+const applyConnect = connect(
+	( state ) => {
+		return {
+			postId: getCurrentPostId( state ),
+			parent: getEditedPostAttribute( state, 'parent' ),
+		};
+	},
+	{
+		onUpdateParent( parent ) {
+			return editPost( { parent: parent || 0 } );
+		},
+	}
+);
+
+const applyWithAPIData = withAPIData( ( props ) => {
+	const restBase = get( props, 'postType.rest_base' );
+	const queryString = stringify( {
+		context: 'edit',
+		per_page: 100,
```

Comment:
Not specific to this PR, but at some point, we'd have to build a paginated "Select" component to allow fetching more items when we reach the end.

Reply:
Yes, I agree, is something we will need to look into not just here but in some other places.